### ISSUE TITLE
Should use uppper case when encoding

### ISSUE THREAD

**Issue by danny-cheung** [agilord/ulid#6]
https://en.wikipedia.org/wiki/Base32#Crockford's_Base32 suggests that the encoded string should only use upper case characters.

**Comment by danny-cheung** [comment]
Not sure if this would be a breaking change.

**Comment by isoos** [comment]
I think that'd be a breaking change now, and for very little benefit. Maybe adding an extra method or a parameter to them?

**Comment by danny-cheung** [comment]
I disagree with the "very little benefit" part. By using lower case, the output is non-compliant and therefore incorrect.

**Comment by isoos** [comment]
> I disagree with the "very little benefit" part. By using lower case, the output is non-compliant and therefore incorrect.

#7 would now break all the codes that rely on the lowercase ids that this code generates. I know at least two apps that use it such way, and their database is case sensitive.

If the code is only soft-compliant, I am willing to live with it.

**Comment by xeard** [comment]
Lowercase is better for the eye. Going with the standard uppercase could be introduced as an option when generating the ids.

**Comment by isoos** [comment]
Added `toBase32` and deprecated `toCanonical`. A future 3.0.0 version will remove `toCanonical` leaving the library spec-aligned. (If there is any other suggestion in this regard, please let me know).
